arelle/utils/validate/DetectScripts.py

```diff
@@ -0,0 +1,101 @@
+from typing import Any, Collection
+
+from arelle.XbrlConst import xhtml
+from lxml.etree import _Element
+
+HTML_EVENT_HANDLER_ATTRIBUTES = frozenset((
+    "onabort",
+    "onafterprint",
+    "onbeforeprint",
+    "onbeforeunload",
+    "onblur",
+    "oncanplay",
+    "oncanplaythrough",
+    "onchange",
+    "onclick",
+    "oncontextmenu",
+    "oncopy",
+    "oncuechange",
+    "oncut",
+    "ondblclick",
+    "ondrag",
+    "ondragend",
+    "ondragenter",
+    "ondragleave",
+    "ondragover",
+    "ondragstart",
+    "ondrop",
+    "ondurationchange",
+    "onemptied",
+    "onended",
+    "onerror",
+    "onfocus",
+    "onhashchange",
+    "oninput",
+    "oninvalid",
+    "onkeydown",
+    "onkeypress",
+    "onkeyup",
+    "onload",
+    "onloadeddata",
+    "onloadedmetadata",
+    "onloadstart",
+    "onmessage",
+    "onmousedown",
+    "onmousemove",
+    "onmouseout",
+    "onmouseover",
+    "onmouseup",
+    "onmousewheel",
+    "onoffline",
+    "ononline",
+    "onpagehide",
+    "onpageshow",
+    "onpaste",
+    "onpause",
+    "onplay",
+    "onplaying",
+    "onpopstate",
+    "onprogress",
+    "onratechange",
+    "onreset",
+    "onresize",
+    "onscroll",
+    "onsearch",
+    "onseeked",
+    "onseeking",
+    "onselect",
+    "onstalled",
+    "onstorage",
+    "onsubmit",
+    "onsuspend",
+    "ontimeupdate",
+    "ontoggle",
+    "onunload",
+    "onvolumechange",
+    "onwaiting",
+    "onwheel",
+))
+
+
+def hasEventHandlerAttributes(elt: Any) -> bool:
+    return _hasEventAttributes(elt, HTML_EVENT_HANDLER_ATTRIBUTES)
+
+
+def _hasEventAttributes(elt: Any, attributes: Collection[str]) -> bool:
+    if isinstance(elt, _Element):
+        return any(a in attributes for a in elt.keys())
+    return False
+
+
+def checkForScript(elt: Any) -> Any:
```

Comment:
I don't know naming conventions, but it seems like `Html` in the filename might be useful, and `contains` seems like a more useful verb than `checkFor`.

It seems like the method could could just return bool.